src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs

```diff
@@ -361,6 +361,27 @@ public static void PrepareDelegate(Delegate d)
 
             return RuntimeImports.RhNewObject(mt);
         }
+
+        public static unsafe object? Box(ref byte target, RuntimeTypeHandle type)
+        {
+            if (type.IsNull)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.type);
+
+            MethodTable* eeType = type.ToMethodTable();
+
+            if (eeType->IsByRef || eeType->IsPointer || eeType->IsFunctionPointer)
```

Comment:
Might want to also add a test for this.

```suggestion
            if (eeType->IsGenericTypeDefinition || eeType->IsByRef || eeType->IsPointer || eeType->IsFunctionPointer)
```

Reply:
Also, System.Void?

Reply:
@MichalStrehovsky Do we need to handle [this case](https://github.com/dotnet/runtime/blob/95b67ed92ee5b785f6dd92efad2404583e05a8e7/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs#L343-L347) here as well?

Reply:
> @MichalStrehovsky Do we need to handle [this case](https://github.com/dotnet/runtime/blob/95b67ed92ee5b785f6dd92efad2404583e05a8e7/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.NativeAot.cs#L343-L347) here as well?

Yes, that too.